common/types/block.go

```diff
@@ -39,47 +40,107 @@ func (id BlockID) AsHash32() Hash32 {
 	return Hash20(id).ToHash32()
 }
 
-var layersPerEpoch int32
+var (
+	layersPerEpoch uint32
+	// effectiveGenesis marks when actual blocks would start being crated in the network, this will take account the first
+	// genesis epoch and the following epoch in which ATXs are published
+	effectiveGenesis uint32
+)
 
-// EffectiveGenesis marks when actual blocks would start being crated in the network, this will take account the first
-// genesis epoch and the following epoch in which ATXs are published
-var EffectiveGenesis int32
+// SetLayersPerEpoch sets global parameter of layers per epoch, all conversion from layer to epoch use this param
+func SetLayersPerEpoch(layers uint32) {
+	atomic.StoreUint32(&layersPerEpoch, layers)
+	atomic.StoreUint32(&effectiveGenesis, layers*2-1)
+}
 
-func getLayersPerEpoch() int32 {
-	return atomic.LoadInt32(&layersPerEpoch)
+func getLayersPerEpoch() uint32 {
+	return atomic.LoadUint32(&layersPerEpoch)
 }
 
-// SetLayersPerEpoch sets global parameter of layers per epoch, all conversion from layer to epoch use this param
-func SetLayersPerEpoch(layers int32) {
-	atomic.StoreInt32(&layersPerEpoch, layers)
-	atomic.StoreInt32(&EffectiveGenesis, layers*2-1)
+// GetEffectiveGenesis returns when actual blocks would be created
+func GetEffectiveGenesis() LayerID {
+	return NewLayerID(atomic.LoadUint32(&effectiveGenesis))
 }
 
-// LayerID is a uint64 representing a layer number. It is zero-based.
-type LayerID uint64
+// NewLayerID creates LayerID from uint32.
+func NewLayerID(value uint32) LayerID {
+	return LayerID{Value: value}
+}
+
+// LayerID is representing a layer number. Zero value is safe to use, and means 0.
+// Internally it is a simple wrapper over uint32 and should be considered immutable
+// the same way as any integer.
+type LayerID struct {
+	Value uint32
```

Comment:
Let's add a comment on this field explaining why it's exported and advising developers not to modify it directly.